test/parallel/test-fs-rm.js

```diff
@@ -280,3 +282,44 @@ function removeAsync(dir) {
     message: /^The value of "options\.maxRetries" is out of range\./
   });
 }
+
+{
+  if (!common.isIBMi && (common.isWindows || process.getuid() !== 0)) {
+    function makeDirectoryReadOnly(dir) {
+      let accessErrorCode = 'EACCES';
+      if (common.isWindows) {
+        accessErrorCode = 'EPERM';
+        execSync(`icacls ${dir} /inheritance:r`);
+        execSync(`icacls ${dir} /deny "everyone":W`);
```

Comment:
```suggestion
        execSync(`icacls ${dir} /deny "everyone:(OI)(CI)(DE,DC)"`);
```
Maybe we need to explicitly set the delete permissions?